### PR TITLE
Update Configuration Variables section

### PR DESCRIPTION
**Pull request by tjorim**
**Description:**
- Updates the Configuration Variables section of several components according to the docs: https://developers.home-assistant.io/docs/en/documentation_create_page.html
- Fix some types to be in line with the rest
- Some other small typo's

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by tjorim** [comment]
Would like to know the answer to https://github.com/home-assistant/home-assistant/pull/15143#issuecomment-408599718 before merging, to reflect correct options in the documentation.

File: source/_components/binary_sensor.ihc.markdown [hunk 1]
```diff
@@ -52,8 +52,8 @@ binary_sensors:
     inverting:
       description: If True the sensor will be inverted.
       required: false
-      type: bool
-      default: false
+      type: boolean
+      default: False
```
Comment (dale3h) [inline source/_components/binary_sensor.ihc.markdown]: Please change to all lowercase (`default: false`).

File: source/_components/cover.template.markdown [hunk 2]
```diff
@@ -87,13 +84,13 @@ cover:
       optimistic:
         description: Force cover position to use [optimistic mode](#optimistic-mode).
         required: false
-        type: bool
-        default: false
+        type: boolean
+        default: False
```
Comment (dale3h) [inline source/_components/cover.template.markdown]: Please change to all lowercase (`default: false`).

File: source/_components/cover.template.markdown [hunk 3]
```diff
@@ -87,13 +84,13 @@ cover:
...
       tilt_optimistic:
         description: Force cover tilt position to use [optimistic mode](#optimistic-mode).
         required: false
-        type: bool
-        default: false
+        type: boolean
+        default: False
```
Comment (dale3h) [inline source/_components/cover.template.markdown]: Please change to all lowercase (`default: false`).

File: source/_components/cover.template.markdown [hunk 4]
```diff
@@ -102,36 +99,24 @@ cover:
 
 ## {% linkable_title Considerations %}
 
-If you are using the state of a platform that takes extra time to load, the
-Template Cover may get an `unknown` state during startup. This results
-in error messages in your log file until that platform has completed loading.
+If you are using the state of a platform that takes extra time to load, the Template Cover may get an `unknown` state during startup. This results in error messages in your log file until that platform has completed loading.
```
Comment (dale3h) [inline source/_components/cover.template.markdown]: Please revert back to <=80 characters per line.

File: source/_components/cover.template.markdown [hunk 5]
```diff
@@ -102,36 +99,24 @@ cover:
...
 If you use `is_state()` function in your template, you can avoid this situation.
 For example, you would replace
 {% raw %}`{{ states.switch.source.state == 'on' }}`{% endraw %}
-with this equivalent that returns `true`/`false` and never gives an unknown
-result:
+with this equivalent that returns `true`/`false` and never gives an unknown result:
```
Comment (dale3h) [inline source/_components/cover.template.markdown]: Please revert back to <=80 characters per line.

File: source/_components/cover.template.markdown [hunk 6]
```diff
@@ -102,36 +99,24 @@ cover:
...
-[`position_template`](#position_template) are specified. Note that this is
-unlikely to be very reliable without some feedback mechanism, since there is
-otherwise no way to know if the cover is moving properly. The cover can be
-forced into optimistic mode by using the [`optimistic`](#optimistic)
-attribute. There is an equivalent mode for `tilt_position` that is enabled
-when [`tilt_template`](#tilt_template) is not specified or when the
-[`tilt_optimistic`](#tilt_optimistic) attribute is used.
+In optimistic mode, the cover position state is maintained internally. This mode is automatically enabled if neither [`value_template`](#value_template) or [`position_template`](#position_template) are specified. Note that this is unlikely to be very reliable without some feedback mechanism, since there is otherwise no way to know if the cover is moving properly. The cover can be forced into optimistic mode by using the [`optimistic`](#optimistic) attribute. There is an equivalent mode for `tilt_position` that is enabled when [`tilt_template`](#tilt_template) is not specified or when the [`tilt_optimistic`](#tilt_optimistic) attribute is used.
```
Comment (dale3h) [inline source/_components/cover.template.markdown]: Please revert back to <=80 characters per line.

File: source/_components/cover.template.markdown [hunk 7]
```diff
@@ -102,36 +99,24 @@ cover:
...
 
 ## {% linkable_title Examples %}
 
 In this section you will find some real life examples of how to use this cover.
 
 ### {% linkable_title Garage Door %}
 
-This example converts a garage door with a controllable switch and position
-sensor into a cover.
+This example converts a garage door with a controllable switch and position sensor into a cover.
```
Comment (dale3h) [inline source/_components/cover.template.markdown]: Please revert back to <=80 characters per line.

**Review by dale3h — CHANGES_REQUESTED** [review]
- Please change all `default: True` to `default: true` (preferred for YAML)
- Please change all `default: False` to `default: false` (preferred for YAML)
- Please reduce all lines to 80 characters or less where possible. To reduce lines with URLs, it is preferred to use reference links in the entire document.

This is just a preliminary review. Another review will be conducted after these changes have been addressed.

**Comment by frenck** [comment]
@tjorim, I really like the effort you are putting off here.
Nevertheless, the PR becomes really big. Almost too big to handle.

Please use smaller PR's., those are easier to move forward.

**Comment by tjorim** [comment]
@frenck You're probably right, I'm sorry. Is it ok if I just stop now?

**Comment by tjorim** [comment]
New PR: https://github.com/home-assistant/home-assistant.io/pull/5929

**Pull request closed without merging**